packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -215,11 +219,20 @@ export async function* OpenAICompletionModel(
 }
 
 export async function* OpenAIChatModel(
-  props: ModelPropsWithChildren & { model: ValidChatModel; logitBias?: Record<string, number> },
+  props: ModelPropsWithChildren & {
+    model: ValidChatModel;
+    logitBias?: Record<string, number>;
+    functionDefinitions?: FunctionDefinition[];
+  },
   { render, getContext, logger }: LLMx.ComponentContext
 ) {
   const messageElements = await render(props.children, {
-    stop: (e) => e.tag == SystemMessage || e.tag == UserMessage || e.tag == AssistantMessage,
+    stop: (e) =>
+      e.tag == SystemMessage ||
+      e.tag == UserMessage ||
+      e.tag == AssistantMessage ||
+      e.tag == FunctionCall ||
+      e.tag == FunctionResponse,
```

Comment:
At this point, maybe we should switch to `[SystemMessage, UserMessage /* ... etc */].includes(e.tag)`, but no big deal.

Reply:
That gives me some type mismatch error between Component<any> and some complex inferred type. I leave it for now :)

Reply:
Great. It would be a waste to spend more than 10 seconds trying to figure it out. :smile: